languages/julia/reference/progression.md

```diff
@@ -53,25 +53,25 @@ I think of closely coupled concepts/exercises as a “block” in the progressio
 - Based on the existing exercises _robot-name_ and _robot-simulator_
 - Might be split up in multiple exercises
 - What is the student supposed to know after solving this exercise?
-    1. How to define an immutable struct `NamedRobot` with field `name::String`
-        - Provide Random-Name-Generator in hints but allow the student to implement their own
-        - **Q:** Why should fields have a concrete type?
-            - Add `@inferred` tests
-    1. How to define a method that acts on the struct `name(r::NamedRobot)`
-    1. How to define a mutable struct `MovingRobot` (?) with fields that describe the position
-        - **PREV:** the position could be a `Complex{Int}`, a `(Named)Tuple{Int,Int}`, two fields `x`, `y` of type `Int`, a struct `Point{Int}`
-        - **NEXT/PREV:** `Point{T<:Number}` could be used to introduce parametric types
-    1. How to define methods that mutate the given argument `move!(r::MovingRobot)`
-        - **Q:** Why does the function have a trailing `!`?
-    1. **NEXT:** Define an abstract type `AbstractRobot` and `NamedRobot <: AbstractRobot`, `MovingRobot <: AbstractRobot`
-    1. **PREV/NEXT:** Wrap it all in a module
+  1. How to define an immutable struct `NamedRobot` with field `name::String`
```

Comment:
This makes the list harder to read unless it's rendered.

(This comment is not part of the requested changes)

Reply:
I'm not seeing much of a difference:



```
- What is the student supposed to know after solving this exercise?
    1. How to define an immutable struct `NamedRobot` with field `name::String`
        - Provide Random-Name-Generator in hints but allow the student to implement their own
        - **Q:** Why should fields have a concrete type?
            - Add `@inferred` tests
    1. How to define a method that acts on the struct `name(r::NamedRobot)`
    1. How to define a mutable struct `MovingRobot` (?) with fields that describe the position
        - **PREV:** the position could be a `Complex{Int}`, a `(Named)Tuple{Int,Int}`, two fields `x`, `y` of type `Int`, a struct `Point{Int}`
        - **NEXT/PREV:** `Point{T<:Number}` could be used to introduce parametric types
    1. How to define methods that mutate the given argument `move!(r::MovingRobot)`
        - **Q:** Why does the function have a trailing `!`?
    1. **NEXT:** Define an abstract type `AbstractRobot` and `NamedRobot <: AbstractRobot`, `MovingRobot <: AbstractRobot`
    1. **PREV/NEXT:** Wrap it all in a module
```

becomes:

```
- What is the student supposed to know after solving this exercise?
  1. How to define an immutable struct `NamedRobot` with field `name::String`
     - Provide Random-Name-Generator in hints but allow the student to implement their own
     - **Q:** Why should fields have a concrete type?
       - Add `@inferred` tests
  1. How to define a method that acts on the struct `name(r::NamedRobot)`
  1. How to define a mutable struct `MovingRobot` (?) with fields that describe the position
     - **PREV:** the position could be a `Complex{Int}`, a `(Named)Tuple{Int,Int}`, two fields `x`, `y` of type `Int`, a struct `Point{Int}`
     - **NEXT/PREV:** `Point{T<:Number}` could be used to introduce parametric types
  1. How to define methods that mutate the given argument `move!(r::MovingRobot)`
     - **Q:** Why does the function have a trailing `!`?
  1. **NEXT:** Define an abstract type `AbstractRobot` and `NamedRobot <: AbstractRobot`, `MovingRobot <: AbstractRobot`
  1. **PREV/NEXT:** Wrap it all in a module
```

So indentation of two spaces instead of four. Would you like me to revert to the previous version?

Reply:
Yes, 4 levels are more readable with that many nested lists, imo.